scvi/dataloaders/_ann_dataloader.py

```diff
@@ -100,62 +33,48 @@ class AnnDataLoader(DataLoader):
         minibatch size to load each iteration
     data_and_attributes
         Dictionary with keys representing keys in data registry (``adata_manager.data_registry``)
-        and value equal to desired numpy loading type (later made into torch tensor).
-        If ``None``, defaults to all registered data.
-    data_loader_kwargs
-        Keyword arguments for :class:`~torch.utils.data.DataLoader`
+        and value equal to desired numpy loading type (later made into torch tensor) or list of
+        such keys. A list can be used to subset to certain keys in the event that more tensors than
+        needed have been registered. If ``None``, defaults to all registered data.
     iter_ndarray
         Whether to iterate over numpy arrays instead of torch tensors
+    data_loader_kwargs
+        Keyword arguments for :class:`~torch.utils.data.DataLoader`
     """
 
     def __init__(
         self,
         adata_manager: AnnDataManager,
-        shuffle=False,
-        indices=None,
-        batch_size=128,
-        data_and_attributes: Optional[dict] = None,
-        drop_last: Union[bool, int] = False,
+        shuffle: bool = False,
+        indices: Union[Sequence[int], Sequence[bool]] = None,
+        batch_size: int = 128,
+        data_and_attributes: Optional[Union[List[str], Dict[str, np.dtype]]] = None,
+        drop_last: bool = False,
         iter_ndarray: bool = False,
         **data_loader_kwargs,
     ):
-        if adata_manager.adata is None:
```

Comment:
surprised that you can delete this w/o any test breaking. it's been too long so I forget why this code is here in the first place

Reply:
nvm i can see it was moved to dataset